genesyscloud/util/util_e164.go

```diff
@@ -5,11 +5,21 @@ import (
 	"github.com/nyaruka/phonenumbers"
 )
 
-func FormatAsE164Number(number string) (string, diag.Diagnostics) {
+// Formats string as a valid E.164 international standard telephone format.
+// Use this when data is coming from the user so we can appropriately error.
+func FormatAsValidE164Number(number string) (string, diag.Diagnostics) {
 	phoneNumber, err := phonenumbers.Parse(number, "US")
 	if err != nil {
 		return "", diag.Errorf("Failed to format phone number %s: %s", number, err)
 	}
 	formattedNum := phonenumbers.Format(phoneNumber, phonenumbers.E164)
 	return formattedNum, nil
 }
+
+// Formats string as a valid E.164 international standard telephone format.
+// Use this function when data is being returned back from the API already in expected format
+func FormatAsCalculatedE164Number(number string) string {
+	phoneNumber, _ := phonenumbers.Parse(number, "US")
```

Comment:
can we have the region retrieved from env variable or aws region with default as "US" ?

Reply:
Hey @HemanthDogiparthi12 , I implemented this by looking up the Organization's Default Country Code and passing that in to the phone numbers library. See commit https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/1178/commits/1ba502536e117a81df0c9627932404c55c601d67
I have tested this commit out against the Japan org and it handles it correctly. I also had to use the dependency injection approach so that I could properly unit test the functions against both "US" and "JP" handling.

I do realize that using a global variable in the providers package is not exactly kosher, but I needed some way to store the Default Country Code. If you have any suggestions of alternative approaches, I'm all ears! Also, if we need to revert this commit in order to get the build out, I'm open to that and revisiting this later if need be.

cc @charliecon @carnellj